packages/server/lib/browsers/webkit-automation.ts

```diff
@@ -322,12 +298,22 @@ export class WebKitAutomation {
    * Clear all cookies
    * @returns cookies cleared
    */
-  private async clearCookies (): Promise<CyCookie[]> {
-    const allCookies = await this.getCookies()
+  private async clearCookies (cookiesToClear: CyCookie[]): Promise<CyCookie[]> {
+    const allCookies = await this.context.cookies()
+    const persistCookies = allCookies.filter((cookie) => {
+      return !cookiesToClear.find((cookieToClear) => {
+        return cookieMatches(cookie, cookieToClear)
+      })
+    })
+
+    debug('clear cookies: %o', cookiesToClear)
+    debug('put back cookies: %o', persistCookies)
 
     await this.context.clearCookies()
 
-    return allCookies
+    if (persistCookies.length) await this.context.addCookies(persistCookies)
```

Comment:
If this is mean to clears all cookies, why would we persist any cookies? does the description need updated? Clears all cookies in domain or clears all the provided cookies?

Reply:
It's meant to clear certain cookies, usually for the domain specified. However, webkit only has a "clear all cookies" method, so we have to clear all the cookies and put back the ones we didn't want cleared. I added comments in https://github.com/cypress-io/cypress/pull/24471/commits/ad2f29435695ab144ff40fed3c85eac3ad1f1cdb to clarify this.